File: accounting.py
wealth_map = {}
n, q = [int(i) for i in input().split()]

restart = 0
for i in range(q):
    args = input().split()
    if args[0] == "SET":
        wealth_map[args[1]] = args[2]
    elif args[0] == "RESTART":
        restart = args[1]
        wealth_map = {}
    elif args[0] == "PRINT":
        try:
            print(wealth_map[args[1]])
        except:
            print(restart)

File: accounting.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int n, q;
  cin >> n >> q;

  vector<int> wealth_map;
  wealth_map.reserve(n);

  int restart = 0;
  while (q--) {
    string event;
    cin >> event;

    if (event == "SET") {
      int i, x;
      cin >> i >> x;
      wealth_map[i - 1] = x;
    } else if (event == "RESTART") {
      int x;
      cin >> x;
      restart = x;
      wealth_map.clear();
    } else if (event == "PRINT") {
      int i;
      cin >> i;
      if (wealth_map[i - 1] == 0) {
        cout << restart << '\n';
      } else {
        cout << wealth_map[i - 1] << '\n';
      }
    }
  }

  return 0;
}